src/ui/public/share/lib/url_shortener.js

```diff
@@ -19,34 +19,29 @@
 
 import chrome from '../../chrome';
 import url from 'url';
+import { kfetch } from 'ui/kfetch';
+import { toastNotifications } from 'ui/notify';
 
-export function UrlShortenerProvider(Notifier, $http) {
-  const notify = new Notifier({
-    location: 'Url Shortener'
-  });
-
-  function shortenUrl(absoluteUrl) {
-    const basePath = chrome.getBasePath();
+export function shortenUrl(absoluteUrl) {
+  const basePath = chrome.getBasePath();
 
-    const parsedUrl = url.parse(absoluteUrl);
-    const path = parsedUrl.path.replace(basePath, '');
-    const hash = parsedUrl.hash ? parsedUrl.hash : '';
-    const relativeUrl = path + hash;
+  const parsedUrl = url.parse(absoluteUrl);
+  const path = parsedUrl.path.replace(basePath, '');
+  const hash = parsedUrl.hash ? parsedUrl.hash : '';
+  const relativeUrl = path + hash;
 
-    const formData = { url: relativeUrl };
+  const body = JSON.stringify({ url: relativeUrl });
 
-    return $http.post(`${basePath}/shorten`, formData).then((result) => {
-      return url.format({
-        protocol: parsedUrl.protocol,
-        host: parsedUrl.host,
-        pathname: `${basePath}/goto/${result.data}`
-      });
-    }).catch((response) => {
-      notify.error(response);
+  return kfetch({ method: 'POST', 'pathname': '/api/shorten_url', body }).then((resp) => {
```

Comment:
async/await?

Reply:
if that is preferred. Figured with the catch block this was just as easy.

Reply:
I think async/await is more than just easy, I think it's also easier to read and more aligned with non-async code, so I would say I prefer it regardless.